xlive/Blam/Engine/saved_games/saved_game_files_async_windows.cpp

```diff
@@ -0,0 +1,252 @@
+{
+	return INVOKE(0x9AE4C, 0, saved_games_async_helper_create_task_write_bin_file, full_path, buffer, header_size, data, data_size, unk, completion);
+}
+
+bool __cdecl saved_games_async_helper_create_task_read_bin_file(wchar_t* full_path, int8* buffer, uint32 header_size, int8* data, uint32 data_size, bool unk, int8* completion)
+{
+	return INVOKE(0x9B67D, 0, saved_games_async_helper_create_task_read_bin_file, full_path, buffer, header_size, data, data_size, unk, completion);
+}
+
+/* public code */
+
+void saved_games_async_helper_get_saved_game_bin_path(uint32 enumerated_file_index, const wchar_t* binary_name, wchar_t* out_path)
+{
+	s_saved_game_main_menu_globals_save_file_info file_info{};
+	saved_games_get_file_info(&file_info, enumerated_file_index);
+
+	wcsncpy(out_path, file_info.file_path, 256);
+	wcscat(out_path, binary_name);
+	wcscat(out_path, L".bin");
+}
+
+bool saved_games_async_helper_check_saved_game_bin(const wchar_t* binary_name, uint32 enumerated_file_index)
+{
+	wchar_t bin_path[MAX_PATH]{};
+
+	saved_games_async_helper_get_saved_game_bin_path(enumerated_file_index, binary_name, bin_path);
+
+	utf8 utf8_path[MAX_PATH];
+	wchar_string_to_utf8_string(bin_path, utf8_path, MAX_PATH);
+
+	s_file_reference file_reference;
+	if (file_reference_create_from_path(&file_reference, utf8_path, false))
+	{
+		e_file_open_error t_error;
+		if (file_open(&file_reference, _permission_read_bit, &t_error))
+		{
+			file_close(&file_reference);
+			return true;
+		}
+	}
+	return false;
+}
+
+bool saved_games_async_helper_write_saved_game_bin(const wchar_t* binary_name, uint32 enumerated_file_index, int8* buffer, uint32 buffer_size)
+{
+	int8 in_out_completion[2060];
+
+	wchar_t bin_path[MAX_PATH]{};
+
+	saved_games_async_helper_get_saved_game_bin_path(enumerated_file_index, binary_name, bin_path);
+
+	bool result = saved_games_async_helper_create_task_write_bin_file(bin_path, (int8*)buffer, 0, (int8*)buffer, buffer_size, 0, in_out_completion);
+
+	async_yield_until_done(in_out_completion, true);
+
+	return result;
+}
+
+bool saved_games_async_helper_read_saved_game_bin(const wchar_t* binary_name, uint32 enumerated_file_index, int8* buffer, uint32 buffer_size)
+{
+	if (saved_games_async_helper_check_saved_game_bin(binary_name, enumerated_file_index))
+	{
+		int8 in_out_completion[2060];
+
+		wchar_t bin_path[MAX_PATH]{};
+
+		saved_games_async_helper_get_saved_game_bin_path(enumerated_file_index, binary_name, bin_path);
+
+		bool result = saved_games_async_helper_create_task_read_bin_file(bin_path, buffer, 0, buffer, buffer_size, 0, in_out_completion);
+
+		async_yield_until_done(in_out_completion, true);
+
+		return result;
+	}
+	return false;
+}
+
+void saved_games_async_helpers_apply_hooks()
+{
+	//DETOUR_ATTACH(p_saved_games_async_helper_write_file, Memory::GetAddress<t_saved_games_async_helper_write_file>(0x42CA6), saved_games_async_helper_write_file);
```

Comment:
?

Reply:
Hooks are not needed but left code and hooks in case they need to be added in the future.